ipykernel/thread.py

```diff
@@ -0,0 +1,40 @@
+"""Base class for threads."""
+from threading import Event, Thread
+
+from anyio import create_task_group, run, to_thread
+
+
+class BaseThread(Thread):
+    """Base class for threads."""
+
```

Comment:
Misc notes – not requests for changes, It's just a long PR and I want to have notes for myself.

Should we use the opportunity to add type annotations ?

Reply:
I added type annotations where it helped, such as some of the new `SubshellManager`. For `Thread` classes I followed the existing code (e.g. https://github.com/ipython/ipykernel/blob/main/ipykernel/control.py) and aimed for minimal changes, so no type annotations.

I am happy to add annotations for the code I've added and/or changed.

Reply:
More type annotations in 5715dd09. I've held back from using `Callable` as that eventually caused me more trouble that I was prepared to deal with at this stage.